code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AndroidUIService.java

```diff
@@ -241,9 +242,11 @@ public void showLocalNotification(final NotificationSetting notificationSetting)
         intent.putExtra(NOTIFICATION_REQUEST_CODE_KEY, requestCode);
         intent.putExtra(NOTIFICATION_DEEPLINK_KEY, notificationSetting.getDeeplink());
         intent.putExtra(NOTIFICATION_CONTENT_KEY, notificationSetting.getContent());
-        intent.putExtra(
-                NOTIFICATION_USER_INFO_KEY,
-                (HashMap<String, Object>) notificationSetting.getUserInfo());
+        final HashMap<String, Object> userInfo =
+                !MapUtils.isNullOrEmpty(notificationSetting.getUserInfo())
+                        ? new HashMap<>(notificationSetting.getUserInfo())
+                        : null;
+        intent.putExtra(NOTIFICATION_USER_INFO_KEY, userInfo);
```

Comment:
Which component creates the `NotificationSettings` in the case of an error? Is this the customer application or the extension?
I ask because the `userInfo` does not seem to have a contract that says the contents are immutable (I understand that they rightly need to be immutable). So I am wondering if this issue can be fixed at the component that creates this setting by using a mutable map instead.

Reply:
Based on internal discussion, this may be needed for more than one extension. I am also inclined with adding it in core.